syntax = "proto3";

package health;

option go_package = "github.com/Shine-5705/WhatStyle_WhatsAppInYourwStyle/pkg/grpc/health";

service HealthService {
    rpc Check(HealthCheckRequest) returns (HealthCheckResponse);
    rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
}

message HealthCheckRequest {
    string service = 1;
}

message HealthCheckResponse {
    ServingStatus status = 1;
    string message = 2;
}

enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
    SERVICE_UNKNOWN = 3;
}